### PR TITLE
First draft of /docs

### PR DESCRIPTION
**Pull request by pbrisbin**
- Move DemoR to DocsR
- Add highlight.js
- Write /docs content (suggestions welcome!)
- Add link to navigation bar

### CODE REVIEW

**Comment by pbrisbin** [comment]
If you'd prefer, I can make the first commit its own PR (but it's needed for this branch to pass CI) and move the second one directly to master.

File: .halcyon/pre-build-hook [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+#
+# Ensure Settings/StaticFiles.hs is always compiled. This is overly-aggressive
+# but is known to work. Coming changes in Halcyon will make this better.
+#
+# Related: https://github.com/mietek/halcyon/issues/53#issuecomment-82392022
+#
+###
```
Comment (gabebw) [inline .halcyon/pre-build-hook]: What's this? Why `###`?

Comment (pbrisbin) [inline .halcyon/pre-build-hook]: This is just how I format comment headers.

```sh
#shebang
#
# Docs...
#
###
code
```

File: templates/docs.hamlet [hunk 2]
```diff
@@ -0,0 +1,121 @@
+<h1>Carnival Documentation
+
```
Comment (gabebw) [inline templates/docs.hamlet]: Can we add an introduction? What's Carnival? How does it work, at a high-level? Let's make a page that people can read without looking at the README.

File: templates/docs.hamlet [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    <tr>
+      <td><code>article_selector</code>
+      <td><code>"article"</code>
+      <td>The CSS selector used to find the Article element.
+    <tr>
+      <td><code>block_selector</code>
+      <td><code>":scope > p, :scope > pre"</code>
+      <td>
+        The CSS selector used to find elements within the Article on which to
+        add comments Threads.
```
Comment (gabebw) [inline templates/docs.hamlet]: comment (singular)?

File: templates/docs.hamlet [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      <td><code>article_author</code>
+      <td><code>""</code>
+      <td>The Article author. Used in the Comments Feed.
+    <tr>
+      <td><code>article_selector</code>
+      <td><code>"article"</code>
+      <td>The CSS selector used to find the Article element.
+    <tr>
+      <td><code>block_selector</code>
+      <td><code>":scope > p, :scope > pre"</code>
```
Comment (gabebw) [inline templates/docs.hamlet]: What's `:scope`? The article selector?

Comment (pbrisbin) [inline templates/docs.hamlet]: Yeah, I guess it's a CSS thing? I don't really know. I assume it means that:

```js
element.querySelector(':scope >p')
```

Finds all direct `p`s under `element`.

**Comment by pbrisbin** [comment]
Added note about :scope and some introductory copy.

**Comment by pbrisbin** [comment]
I really need to add a `$(markdownFile)` function, writing HTML kinda sucks...

File: .halcyon/pre-build-hook (hunk 1 above)
Comment (gabebw) [inline .halcyon/pre-build-hook]: Sure.

File: templates/docs.hamlet (hunk 4 above)
Comment (gabebw) [inline templates/docs.hamlet]: Oh, neat: https://developer.mozilla.org/en-US/docs/Web/CSS/:scope

How about linking to those docs?

File: templates/docs.hamlet [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  <p>
+
+    By default, a hidden side-bar will be added to any 
+    <code>article
+    elements. Within these elements, each 
+    <code>p
+    and 
+    <code>pre
+    element will receive a small indicator, visible on hover. When clicked, the
+    side-bar will expand allowing visitors to comment on that specific portion
```
Comment (gabebw) [inline templates/docs.hamlet]: sidebar?

Comment (gabebw) [inline templates/docs.hamlet]: expand, allowing

File: templates/docs.hamlet [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    of your article.
+
+  <p>
+
+    Carnival also sends email notifications on replies and provides an RSS feed
+    of recent comments on your Site.
+
+  <p>
+
+    This page itself has Carnival loaded. Feel free to add comments and
```
Comment (gabebw) [inline templates/docs.hamlet]: Niiiice.

File: templates/docs.hamlet [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    .
+
+  <p>
+
+    When comments are opened, Carnival will add a 
+    <code>commenting
+    class to your Article element. To make room for itself, Carnival will shift
+    your Article element 
+    <code>150
+    pixels to its left. It does this by taking the element's current
```
Comment (gabebw) [inline templates/docs.hamlet]: Maybe "to the left"? Or is this text [irreplaceable](https://www.youtube.com/watch?v=2EwViQxSJJQ)?

File: templates/docs.hamlet (hunk 4 above)
Comment (pbrisbin) [inline templates/docs.hamlet]: Done.

File: templates/docs.hamlet [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  <p>
+    Carnival is open-source software. You can directly add issues and track its
+    development on 
+    <a href="https://github.com/thoughtbot/carnival">GitHub
+    .
+
+  <p>
+    You can also email questions and comments to 
+    <a href="mailto:support@carnivalapp.io">
+      carnival-support@thoughtbot.com
```
Comment (gabebw) [inline templates/docs.hamlet]: Why is the mailto href different from what we display?

Comment (pbrisbin) [inline templates/docs.hamlet]: Because I screwed up...

File: templates/docs.hamlet [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    .
+
+<!-- Hey, look how easy this is! -->
```
Comment (gabebw) [inline templates/docs.hamlet]: Cute!

**Comment by gabebw** [comment]
:+1:

**Pull request merged**